extensions/resource-deployment/src/utils.ts

```diff
@@ -30,12 +30,18 @@ export function setEnvironmentVariablesForInstallPaths(tools: ITool[]): void {
 			// construct an env variable name with NoteBookEnvironmentVariablePrefix prefix
 			// and tool.name as suffix, making sure of using all uppercase characters and only _ as separator
 			const envVarName = getRuntimeBinaryPathEnvironmentVariableName(t.name);
-			process.env[envVarName] = t.installationPathOrAdditionalInformation;
+			env[envVarName] = t.installationPathOrAdditionalInformation;
 			installationPaths.add(path.dirname(t.installationPathOrAdditionalInformation));
 		}
 	});
 	if (installationPaths.size > 0) {
 		const envVarToolsInstallationPath: string = [...installationPaths.values()].join(path.delimiter);
-		process.env[ToolsInstallPath] = envVarToolsInstallationPath;
+		env[ToolsInstallPath] = envVarToolsInstallationPath;
+	}
+}
+
+export function assert(condition: boolean, message?: string): asserts condition {
```

Comment:
what does the 'asserts condition' at the end means

Reply:
This means wherever this assert function is called it acts a type guard check corresponding to the 'condition' expression.